airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -123,103 +131,98 @@ protected void startTracked() throws Exception {
     pairToIgnoredRecordCount.clear();
     LOGGER.info("{} started.", BufferedStreamConsumer.class);
 
-    LOGGER.info("Buffer creation started for {} streams.", pairs.size());
-    final Path queueRoot = Files.createTempDirectory("queues");
-    for (AirbyteStreamNameNamespacePair pair : pairs) {
-      LOGGER.info("Buffer creation for stream {}.", pair);
-      try {
-        final OnDiskQueue writeBuffer = new OnDiskQueue(queueRoot.resolve(pair.getName()), pair.getName());
-        pairToWriteBuffer.put(pair, writeBuffer);
-      } catch (Exception e) {
-        LOGGER.error("Error creating buffer: ", e);
-      }
-    }
-    LOGGER.info("Buffer creation completed.");
-
     onStart.call();
-    LOGGER.info("write buffers: {}", pairToWriteBuffer.keySet());
-    writerPool.scheduleWithFixedDelay(
-        () -> writeStreamsWithNRecords(MIN_RECORDS, pairToWriteBuffer, recordWriter),
-        THREAD_DELAY_MILLIS,
-        THREAD_DELAY_MILLIS,
-        TimeUnit.MILLISECONDS);
   }
 
   @Override
   protected void acceptTracked(AirbyteMessage message) throws Exception {
     Preconditions.checkState(hasStarted, "Cannot accept records until consumer has started");
 
-    if (message.getType() != AirbyteMessage.Type.RECORD) {
-      return;
+    if (message.getType() == Type.RECORD) {
+      final AirbyteRecordMessage recordMessage = message.getRecord();
+      final AirbyteStreamNameNamespacePair stream = AirbyteStreamNameNamespacePair.fromRecordMessage(recordMessage);
+      final String data = Jsons.serialize(message);
+
+      if (!pairs.contains(stream)) {
+        throwUnrecognizedStream(catalog, message);
+      }
+
+      if (!isValidRecord.apply(data)) {
+        pairToIgnoredRecordCount.put(stream, pairToIgnoredRecordCount.getOrDefault(stream, 0L) + 1L);
+        return;
+      }
+
+      if (queue.size() == BATCH_SIZE) {
```

Comment:
It better to flush as soon as possible instead of waiting for the next call to accept. So I would move that test after `offer`